### PR TITLE
[frontend] Add and configure cuix library

### PR DESCRIPTION
**Pull request by bjornalm**
## What changes were proposed in this pull request?

- Added and installed _cuix_ and _cuix-core_ as local tgz files 
  The cuix and cuix-core packages have been modified to fit open source usage. A **custom node app** has been created to help with the generation of such modified packages when cuix and cuix-core packages needs to be updated. This app will not be published as open source.
- Updated contributing-frontend.md with a new section about our component libraries (cuix & antd)
- Imported cuix styles to Hue.less
- Removed duplicated sass variables that are now directly available via cuix-core
- Updated the ReactExample.tsx component to use a button from cuix and an icon from cuix-core

## TODOs
1. A new PR with more sass cleanup (the code is ready, will push PR once this one is merged)
2. Solve Jest issue where the ESM (ECMAScript module) exports from cuix and cuix-core can be handled (Jest expects CommonJS modules). There are ways to achieve this but I think the complexity of Hue adds additional challenges. For now the cuix and cuix-core components are mocked in the ReactExample.test.tsx

## How was this patch tested?
- Manual testing of the ReactExample.tsx component

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

File: contributing-frontend.md [hunk 1]
```diff
@@ -72,39 +78,99 @@ const { t, ready } = i18nReact.useTranslation(undefined, { useSuspense: false })
...
 
-
-
 ### Application wide communication
+
 For communication with non react based parts of Hue there is a publish–subscribe messaging system called [huePubsub](https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/js/utils/huePubSub.ts). To publish a message, call the `publish` function with a topic. You can subscribe to a topic using the hook [usePubSub](https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/js/reactComponents/useHuePubSub.ts) which will force your functional component to rerender once a message matching the topic is published. In the example below the editCursor state is updated by useHuePubSub each time a CURSOR_POSITION_CHANGED_EVENT is published.
 
 ```js
-const editorCursor = useHuePubSub<EditorCursor>({ topic: CURSOR_POSITION_CHANGED_EVENT });
+const editorCursor = useHuePubSub < EditorCursor > { topic: CURSOR_POSITION_CHANGED_EVENT };
```
Comment (JohanAhlen) [inline contributing-frontend.md:86]: Nit: Doesn't seem right, the original was good no?

**Review by JohanAhlen — APPROVED** [review]
Nice!

**Review by JohanAhlen — COMMENTED** [review]
You should update the license checker to exclude cuix. It's currently failing as the license is unknown.

File: desktop/core/src/desktop/js/apps/editor/components/result/reactExample/ReactExample.tsx [hunk 2]
```diff
@@ -76,11 +79,17 @@ const ReactExample: FunctionComponent<ReactExampleProps> = ({ activeExecutable,
     // Also make sure that the component specific Antd style is imported in the file
     // 'root-wrapped-antd.less'.
     <React.StrictMode>
-      <div className="react-example antd">
+      <div className="react-example cuix antd">
```
Comment (tabraiz12) [inline desktop/core/src/desktop/js/apps/editor/components/result/reactExample/ReactExample.tsx:82]: is cuix and antd classes compulsory? How will someone new coming into the project know to add this.

**Review by nidhibhatg — APPROVED** [review]

**Review by JohanAhlen — APPROVED** [review]

File: contributing-frontend.md (hunk 1 above)
Comment (bjornalm) [reply contributing-frontend.md]: You are correct. It was Prettier that was being clever and modifying the code example. Changed the code example language from js to ts.

File: desktop/core/src/desktop/js/apps/editor/components/result/reactExample/ReactExample.tsx (hunk 2 above)
Comment (bjornalm) [reply desktop/core/src/desktop/js/apps/editor/components/result/reactExample/ReactExample.tsx:82]: Sorry my previous answered was never saved when I resolved the issue. Yes, but only if the component is a react root. I've updated the contribute frontend instructions.

**Pull request merged**